var/spack/repos/builtin/packages/icu4c/package.py

```diff
@@ -39,11 +40,25 @@ class Icu4c(AutotoolsPackage):
     version('58.2', 'fac212b32b7ec7ab007a12dff1f3aea1')
     version('57.1', '976734806026a4ef8bdd17937c8898b9')
 
+    # The --enable-rpath option is needed on MacOS, but it breaks the
+    # build for xerces-c on Linux.
+    variant('rpath', default=sys.platform == 'darwin',
```

Comment:
To elaborate on https://github.com/spack/spack/pull/9393#issuecomment-428282641, the distinction between this and the `conflicts` directives below is that there is no Spack-interpreted property like `when` to set the Variant's default value, so instead of conditioning on the Spec, you have to use Python's own libraries to set sensible defaults for the Spec.